### PR TITLE
Support special forms in editor hover/completion and simple signature help provider

### LINKED ISSUE #441: [Feature] Support special forms in editor hover/completion
The way to retrieve hover/completion information for special forms like def (see [Special Forms](https://clojure.org/reference/special_forms)) is different from normal symbols like functions. The REPL window has an completion implementation which handles these to cases correctly. This should be implemented in the editor as well. Additionally the editor hover implementation must be extended to show all retrieved information as well.

### PR DESCRIPTION
**Pull request by cfehse**
## What has Changed?

- Added support for special forms and Java static methods in hover and completion. 
- Corrected completion to work if the item to get `info` for does not have a namespace.
- Added a `try/catch` in src/providers/definition.ts` to avoid uncaught exception.
- Moved the parsing of the `info` results into a separate unit.
- Added a simple signature help provider to show the possible parameter list when typing.

simple-signature-provider

Fixes #441, #361 

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Made sure I am directing this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I am changed the default PR base branch, so that it is not `master`. (Sorry for the nagging.)
- [x] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.) You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. (For now you'll need to opt in to the CircleCI _New Experience_ UI to see the Artifacts tab, because bug.)
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.

## The Calva Team PR Checklist:

Before merging we (at least one of us) have:

- [x] Made sure the PR is directed at the `dev` branch (unless reasons).
- [x] Read the source changes.
- [x] Given feedback and guidance on source changes, if needed. (Please consider noting extra nice stuff as well.)
- [x] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.)
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] If need be, had a chat within the team about particular changes.

Ping @pez, @kstehn, @bpringe

### CODE REVIEW

File: src/providers/infoparser.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    private _arglist: string = undefined;
+
+    private _formsString: string = undefined;
+
+    private _docString: string = undefined;
+
+    private _specialForm: boolean = false;
+
+    constructor(msg: any) {
+
```
Comment (PEZ) [inline src/providers/infoparser.ts]: The Calva code base do not use these empty first lines for functions. Nit-picking, of course, but I see it is rather consistent in this new file, so thought it was worth a mention.

**Comment by PEZ** [comment]
I've tested this a while now and gotta say I love it.

File: src/providers/infoparser.ts (hunk 1 above)
Comment (cfehse) [reply src/providers/infoparser.ts]: Changed.

**Comment by cfehse** [comment]
@PEZ, @kstehn, @bpringe

I added a simple signature help provider to show the possible parameters while typing (see description above).

**Pull request merged**